src/flamenco/vm/fd_vm_private.h

```diff
@@ -260,35 +458,40 @@ static inline void fd_vm_mem_st_8( ulong haddr, ulong  val ) { memcpy( (void *)h
 
    FD_VM_MEM_HADDR_LD_FAST and FD_VM_HADDR_ST_FAST are for use when the
    corresponding vaddr region it known to correctly resolve (e.g.  a
-   syscall has already done preflight checks on them). */
-
-#define FD_VM_MEM_HADDR_LD( vm, vaddr, align, sz ) (__extension__({                                       \
-    fd_vm_t const * _vm     = (vm);                                                                       \
-    ulong           _vaddr  = (vaddr);                                                                    \
-    int             _sigbus = _vm->check_align & (!fd_ulong_is_aligned( _vaddr, (align) ));               \
-    ulong           _haddr  = fd_vm_mem_haddr( vm, _vaddr, (sz), _vm->region_haddr, _vm->region_ld_sz, 0, 0UL ); \
-    if( FD_UNLIKELY( (!_haddr) | _sigbus) ) return FD_VM_ERR_SIGSEGV;                                     \
-    (void const *)_haddr;                                                                                 \
+   syscall has already done preflight checks on them).
+   
+   FIXME: These macros currently don't support multi region loads/stores */
```

Comment:
Do we need to change these to support multi-region loads/stores before merging? These are used everywhere inside the syscalls, and a lot of the size values are user-provided, so it's not impossible they could span multiple regions.

Reply:
we don't need to. made a small change to macros to support this

Reply:
Don't we need to error out if `_is_multi` is true? Also can we document why we don't need to support multi-region loads in the syscalls (because Agave's memory translation logic doesn't).

Reply:
I did